Cargo.toml

```
@@ -66,4 +67,7 @@ rpath = false
 strip = "debuginfo"
 
 [features]
+default = ["in_memory_cache", "redis"]
 dhat-heap = ["dep:dhat"] 
+in_memory_cache = ["dep:mini-moka"]
+redis = ["dep:redis"]
```

```suggestion
default = ["memory-cache"]
dhat-heap = ["dep:dhat"] 
memory-cache = ["dep:mini-moka"]
redis-cache = ["dep:redis"]
no-cache = []
hybrid-cache = ["memory-cache", "redis-cache"]
```
I would suggest having two more features, `hybrid-cache` and `no-cache`. Also, I would recommend only having the `memory-cache` to be included as a default feature because as you had discussed before, most people might not require the use of Redis at all and would be satisfied using the `memory-cache` feature. So I think only having the `memory-cache` would be better. What do you say?

---

The no-cache feature wont work, because these features are additive, so if you run:
```
cargo build  --features no-cache
```
That would still have the features specified in the default.
To have a build, without any caching, you need to build with:
```
cargo build  --no-default-features
```

Similarly, 
```
cargo build  --features redis-cache
```
will build with enabled redis-cache AND memory-cache too. 

I've renamed the features

---

Ok I see, thanks :heart: again for pointing this out. But I would like to make one last suggestion I don't know if it is good but I think I will still make it.

My suggestion is that we should take this approach `if it works` because I think this will solve the problem by covering all the four options.

For memory caching solution we can have it like this:

``` shell
cargo build 
```

This will as we know will enable the default feature `memory-cache`.

For hybrid caching solution we can have it like this:

``` shell
cargo build --feature redis-cache
```

For redis caching solution we can have it like this:

``` shell
cargo build --no-default-features --feature redis-cache
```

For no caching solution we can have it like this:

``` shell
cargo build --no-default-features
```

I think this approach will be better because this will eliminate the need to have the config option `in_memory_cache`. Also this will give us 3 advantages one it will reduce binary size which can be beneficial for any hosting on a low storage device, second remove unneccesary code like for example if we have memory caching solution enabled then this will make sure to remove the unneccesary redis cache code from the binary, finally third advantage is that build time will improve which is good you can see. That's why I am actually wanting to have feature based approach. 

Also, the reason why I am suggesting feature based approach is that you see the user will stick to only one caching solution when they will first build the app. It would be a very rare case when the user might move from one caching solution to other. 

Also, finally I think for the inconsistancy and complexity. I think we can explain all this build option in the `docs` which will make it easy for the user to follow and build the app. So with this in mind I don't think it would be a big issue. What do you say? :slight_smiling_face: